### PR TITLE
Fix screenshots

### PR DESCRIPTION
**Pull request by agohorel**
So, remember how I "magically" guessed that passing `false` into p5's `save()` (screenshot) would prevent it from clearing the canvas? I don't know what the hell was happening, but that a) is not valid syntax and was throwing an error and b) even without the `false` it doesn't seem to be clearing the canvas now? 🤔

At any rate, I fixed it, but then realized that with the introduction of chat, it posed an issue where whenever you would type `p`, it would download a screenshot. I thought about adding a flag that would indicate that any `<input>` element was currently in-focus (i.e. you're typing in a text input), and to just ignore the `save` command if that was the case. But I ended up moving towards `ctrl`-based hotkeys to work around that and moved the hotkey to `CTRL` + `x` (export, I guess?). I added a class for tracking which keys are pressed and mapping them to an array of commands. I like this because a) we can move the commands outside of `app.js` and b) we could add more hotkeys, possibly for the actual drawing functions (think photoshop or similar). The only kinda sucky thing is that the browser has a lot of default hotkeys, which is why I didn't make it the more obvious `CTRL` + `p` (print). There might be a way around that, but in any case the commands can be totally arbitrary, so there are infinite combinations of possible commands (they also don't have to be just 2 keys).

### CODE REVIEW

File: public/src/utils/KeyManager.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export class KeyManager {
+  constructor(p5) {
+    this.keysPressed = [];
+    this.commands = [
```
Comment (agohorel) [inline public/src/utils/KeyManager.js:4]: commands will live here - for now you can see each command consists of a `name`, `keys` array, `execute` command, and a single `param` that can optionally be passed to the `execute` function. we could potentially make it so it can accept more than one argument, but for now just keeping it simple.

File: public/src/utils/KeyManager.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      {
+        name: "screenshot",
+        keys: [17, 88], // ctrl + x
+        execute: p5.save.bind(p5),
```
Comment (agohorel) [inline public/src/utils/KeyManager.js:8]: this `bind` usage here is the same deal as in the [`renderShape` function](https://github.com/glitch-dot-cool/paint-with-friends/blob/616ac39b8addb4726597c8d2af7d91f0e51d8ff9/public/src/utils/drawing.js#L30). `bind` is a little confusing, but [this description from MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) summarizes it better than I would be able to: 
> The bind() method creates a new function that, when called, has its [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) keyword set to the provided value

basically, it's just making sure that the function has access to the `p5` object.

File: public/src/app.js [hunk 3]
```diff
@@ -6,9 +6,11 @@ import { LocalStorage } from "./utils/LocalStorage.js";
 import { userList } from "./components/userList.js";
 import { initUsername } from "./utils/initUsername.js";
 import { chatMessages } from "./components/chatMessages.js";
+import { KeyManager } from "./utils/KeyManager.js";
 
 const app = (s) => {
   const socket = io.connect("http://localhost:3000");
+  const keysPressed = new KeyManager(s);
```
Comment (agohorel) [inline public/src/app.js:13]: the `KeyManager` takes in the instance of the `p5` library, so we can do p5 things within it - in this case the `p5.save()` function for screenshotting.

**Review by niceguynaum — APPROVED** [review]
LGTM!
